python_translators/translators/composite_parallel_translator.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+import threading
+
+from python_translators.translators.translator import Translator
+from python_translators.translation_query import TranslationQuery
+from python_translators.translation_response import TranslationResponse
+from python_translators.translation_costs import TranslationCosts
+from python_translators.translation_response import merge_translations
+from python_translators.translators.composite_translator import CompositeTranslator
+
+
+def translate_worker(translator: Translator, query: TranslationQuery, responses: [TranslationResponse], idx: int):
+    responses[idx] = translator.translate(query)
+
+
+class CompositeParallelTranslator(CompositeTranslator):
+    def __init__(self, source_language: str, target_language: str, translators: [Translator] = None):
+        super(CompositeParallelTranslator, self).__init__(source_language, target_language, translators)
+
+    def _translate(self, query: TranslationQuery) -> TranslationResponse:
+        responses = [None] * len(self.translators)
+        threads = []
+
+        # Start a thread for each translator
+        for idx, translator in enumerate(self.translators):
+            t = threading.Thread(target=translate_worker, args=(translator, query, responses, idx))
+            t.start()
+            threads.append(t)
+
+        # Wait for all threads to complete
```

Comment:
here it would be good to be able to specify a `strategy` for the composite translator so it returns before all the sub-translators finish. example strategies: 
1. timeout in ms 
2. N of the translators finished with results 
3. already returned M (distinct) translations 
This third point reminds me that currently we are passing the number of required 
translations as an argument to the translate function. It should probably 
be passed as an argument at the construction time of the translator. Then we 
could implement point 3 in a most natural manner. What do you think?